akka-docs/rst/scala/code/docs/akka/typed/IntroSpec.scala

```diff
@@ -0,0 +1,206 @@
+/**
+ * Copyright (C) 2014 Typesafe Inc. <http://www.typesafe.com>
+ */
+package docs.akka.typed
+
+//#imports
+import akka.typed._
+import akka.typed.ScalaDSL._
+import akka.typed.AskPattern._
+import scala.concurrent.Future
+import scala.concurrent.duration._
+import scala.concurrent.Await
+//#imports
+import akka.testkit.AkkaSpec
+import akka.typed.TypedSpec
+
+object IntroSpec {
+
+  //#hello-world-actor
+  object HelloWorld {
+    final case class Greet(whom: String, replyTo: ActorRef[Greeted])
+    final case class Greeted(whom: String)
+
+    val greeter = Static[Greet] { msg ⇒
+      println(s"Hello ${msg.whom}!")
+      msg.replyTo ! Greeted(msg.whom)
+    }
+  }
+  //#hello-world-actor
+
+  //#counter-actor
+  object Counter {
+
+    sealed trait Command
+    final case class GetAndAdd(amount: Int, replyTo: ActorRef[CurrentCount]) extends Command
+    final case class AddAndGet(amount: Int, replyTo: ActorRef[CurrentCount]) extends Command
+
+    final case class CurrentCount(x: Int)
+
+    val counter: Behavior[Command] = behavior(0)
```

Comment:
WDYM globals? We have many vals in objects, they are immutable values after all. The beauty of this model is that I can start 1000000 counters and they will all share the exact same object for their behavior (until they start counting, but for a stateless Actor this pattern could save a lot of heap space—we could even cache the first 100 counter behaviors if we intend to have LOTS of them running).